### PR TITLE
Dial timeout and tests simplified

### PR DESCRIPTION
**Pull request by philipandag**
This PR adds a timeout to dialing a node. Libp2p gives no way of determining who was being dialed when a dial fails, so it's implemented as a timeout in Node. 

A test for this behaviour was added and because the current tests in bash are very crude I couldn't hold myself back from simplifying them at least a bit. Something like libraries was added alongside an `asserts.sh` "library" which defines functions for basic asserts to save time on writing long if-else checks

### CODE REVIEW

File: liberum_core/src/node/mod.rs
Comment (originalmk) [inline liberum_core/src/node/mod.rs:236]: Why use upper case, it is not constant.

File: tests/dial_nonexisting.sh
Comment (originalmk) [inline tests/dial_nonexisting.sh:18]: Why not loop, which waits for file to be created?

File: tests/dial_nonexisting.sh
Comment (originalmk) [inline tests/dial_nonexisting.sh:34]: What if liberum_cli do not return anything?

File: liberum_core/src/node/mod.rs
Comment (philipandag) [reply liberum_core/src/node/mod.rs]: Actually, as you can see there is no `mut`, so it is a constant :nerd_face: 
https://github.com/Liberum-Neto/liberum-neto/pull/53/commits/aeb075baf393028a6a02a59d951f5a394f2c2598

File: tests/dial_nonexisting.sh
Comment (philipandag) [reply tests/dial_nonexisting.sh:18]: There are a lot of problems with the scripts. Sleep is used here in all the tests and it's not a part of this PR to change the way the tests are generally performed. If keeping the tests in bash is desirable then I'd prefer making a separate issue/PR which will improve them. If we would like to do the tests more professionally then using something like robot framework would be a much better idea. For now a simple sleep works fine.

File: tests/dial_nonexisting.sh
Comment (philipandag) [reply tests/dial_nonexisting.sh]: Well, the test won't pass. Unless you mean the liberum_cli goes dead-locked and hangs. Then the test still won't pass :). Implementing timeouts here is also a subject for improvements in the way all the tests are performed and using a testing framework instead of writing it ourselves from scratch should be discussed

**Review by originalmk — APPROVED** [review]

**Comment by philipandag** [comment]
@originalmk rebase onto master has dismissed your approval. Please redo it

**Review by originalmk — DISMISSED** [review]

**Review by originalmk — APPROVED** [review]

**Pull request merged**
